compiler/passes/flattenFunctions.cpp

```diff
@@ -93,6 +93,14 @@ passableByVal(Type* type) {
 // Otherwise passing by value is more efficient.
 static bool
 passByRef(Symbol* sym) {
+
+  // Is it a non-POD record type?
+  // non-POD records/tuples are passed by reference here because
+  // other code probably won't copy them correctly if they
+  // are passed by value.
+  if( isRecord(sym->type) && !isPOD(sym->type) )
+    return true;
+
```

Comment:
This seems to be a bad change. It makes test/types/records/ferguson/parallel/coforall.chpl (a string-as-rec test) no longer function, because the iterator yields a reference to r, and r is stored in a heap and freed before the task can launch.